### PR TITLE
lint: imports, dependencies, modern type hints, exports

### PR DESCRIPTION
**Pull request by davidlougheed**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -34,7 +34,7 @@ classifiers = [
 python = "^3.10.0"
 bento-lib = {extras = ["django"], version = "^11.3.0"}
 Django = "^4.2.9"
-django-autocomplete-light = "^3.9.4"
+django-autocomplete-light = "^3.11.0rc1"
```
Comment (v-rocheleau) [inline pyproject.toml:37]: Any reason to use a RC version?

**Review by v-rocheleau — COMMENTED** [review]
Lgtm, just a question regarding version change

File: pyproject.toml (hunk 1 above)
Comment (davidlougheed) [reply pyproject.toml:37]: future django 5 support + official Django 4.2 support. they didn't even release a 3.10 non-rc version, so I don't really understand how their versions work - they maybe seem to be misunderstanding semver.

Comment (davidlougheed) [reply pyproject.toml:37]: asked about it here: https://github.com/yourlabs/django-autocomplete-light/issues/1351

**Review by v-rocheleau — APPROVED** [review]
LGTM

**Pull request merged**